File: backend/src/main/java/com/gzouli/ERP/service/car/CarService.java
package com.gzouli.ERP.service.car;

import com.gzouli.ERP.dto.car.CarAssignmentDTO;
import com.gzouli.ERP.dto.car.CarDTO;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDate;
import java.util.List;

public interface CarService {
    List<CarDTO> getAllCars();

    
    CarDTO createCar(CarDTO dto);

    
    CarDTO updateCar(Long id, CarDTO dto);

    
    void deleteCar(Long id);

    
    void assignCar(CarAssignmentDTO request);

    
    void releaseCar(Long carId, LocalDate returnDate);
}

File: frontend/src/app/services/car-service.ts
import { Injectable } from '@angular/core';
import {environment} from '../../environments/environment';
import {Observable} from 'rxjs';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Car} from '../models/car';

@Injectable({
  providedIn: 'root',
})
export class CarService {
  private apiUrl = `${environment.apiUrl}/admin/cars`;

  constructor(private http: HttpClient) {}

  getAllCars(): Observable<Car[]> {
    return this.http.get<Car[]>(this.apiUrl);
  }

  createCar(car: any): Observable<Car> {
    return this.http.post<Car>(this.apiUrl, car);
  }

  updateCar(id: number, car: any): Observable<Car> {
    return this.http.put<Car>(`${this.apiUrl}/${id}`, car);
  }

  deleteCar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  // Assigner un véhicule à un employé
  assignCar(data: { carId: number, employeeId: number, startDate: Date }): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/assign`, data);
  }

  // Récupérer le véhicule (Libération)
  releaseCar(carId: number, returnDate: string): Observable<void> {
    const params = new HttpParams().set('date', returnDate);
    return this.http.patch<void>(`${this.apiUrl}/${carId}/release`, {}, { params });
  }
}
